spec/features/staff/users_can_edit_a_comment_spec.rb

```diff
@@ -103,21 +31,22 @@
       before { authenticate!(user: delivery_partner_user) }
 
       context "when the report is editable" do
-        scenario "the user can edit a comment" do
-          visit organisation_activity_comments_path(activity.organisation, activity)
-          click_on t("table.body.report.edit_comment")
-          fill_in "comment[comment]", with: "Amendments have been made"
-          click_button t("default.button.submit")
-          expect(page).to have_content "Amendments have been made"
+        scenario "the user can edit any comments left by users in the same organisation" do
+          form = CommentForm.edit_from_activity_page(report: report, comment: comment)
+
+          expect(form).to have_report_summary_information
+          form.complete(comment: "Amendments have been made")
```

Comment:
and the `has_report_summary_information?` matcher really pays off in communicating the high level behaviour clearly. Bravo!

I'm going to have a go at reworking those Adjustment features using this style...

Reply:
Nice - I think there's a bunch of places we can use this approach